aspnetcore/security/key-vault-configuration.md

```diff
@@ -0,0 +1,144 @@
+
+App Setting | Description | Example
+--- | --- | ---
+`Vault` | Azure Key Vault name | contosovault
+`ClientId` | Azure Active Directory App Id | 627e911e-43cc-61d4-992e-12db9c81b413
+`ClientSecret` | Azure Active Directory App Key | g58K3dtg59o1Pa+e59v2Tx829w6VxTB2yv9sv/101di=
+
+[!code-csharp[Main](key-vault-configuration/sample/Startup.cs?name=snippet1&highlight=5,10-13)]
+
+`AddAzureKeyVault()` also provides an overload that accepts an implementation of `IKeyVaultSecretManager`, which allows you to control how key vault secrets are converted into configuration keys. For example, the interface can be implemented to load configuration values by application, where you would prefix application names to configuration secrets you've stored in the key vault. This would allow you to maintain secrets for multiple applications in one key vault.
+
+Assume we have several `ConnectionString` key vault secrets with the application name prefixed. For the sample application, we would create a secret in key vault for `KeyVaultConfigProviderSample-ConnectionString` and its value. For a second application, we would create a secret for `SomeOtherApplicationName-ConnectionString` and its value. We would like each app to load its own `ConnectionString` secret into its configuration as `ConnectionString`. An example of this implementation is shown below.
+
+```csharp
+public class EnvironmentSecretManager : IKeyVaultSecretManager
+{
+    private readonly string _appNamePrefix;
+
+    public EnvironmentSecretManager(string appName)
+    {
+        _appNamePrefix = appName + "-";
+    }
+
+    public bool Load(SecretItem secret)
+    {
+        return secret.Identifier.Name.StartsWith(_appNamePrefix);
+    }
+
+    public string GetKey(SecretBundle secret)
+    {
+        return secret.SecretIdentifier.Name.Substring(_appNamePrefix.Length);
+    }
+}
+```
+
+```csharp
+builder.AddAzureKeyVault(
+    $"https://{config["Vault"]}.vault.azure.net/",
+    config["ClientId"],
+    config["ClientSecret"],
+    new EnvironmentSecretManager(env.ApplicationName));
+    
+Configuration = builder.Build();
+```
+
+The `Load()` method is called by an algorithm in the provider that iterates through the secrets to find the one that matches the application name as a prefix to the secret's name. When a match is found with `Load()`, the algorithm uses the `GetKey()` method to return the configuration name of the secret name. It simply strips off the application name prefix from the secret's name and returns the name for loading into the app's configuration name-value pairs.
+
+If you implemented this approach with the sample application, the key vault secrets would be loaded, the string secret for `KeyVaultConfigProviderSample-ConnectionString` would be matched, and the application name `KeyVaultConfigProviderSample` (with the dash) would be stripped off to load `ConnectionString` with its value into the app's configuration.
```

Comment:
consider:

If you implemented this approach with the sample app:
- The key vault secrets would be loaded.
- The string secret for `KeyVaultConfigProviderSample-ConnectionString` would be matched.
- The app name `KeyVaultConfigProviderSample` (with the dash) would be stripped off to load `ConnectionString` with its value into the app's configuration.

Sentence is WAY too long for ESL, our machine translators and me.
The last bullet is too long too.

Is this what you meant?
The app name `KeyVaultConfigProviderSample` (with the dash) would be stripped off **and used** to load `ConnectionString` with its value into the app's configuration.